docs/reference/search/search-your-data/semantic-search-inference.asciidoc

```diff
@@ -0,0 +1,270 @@
+that the model will create based on your input text - must be created. The 
+destination index must have a field with the <<dense-vector, `dense_vector`>> 
+field type to index the output of the OpenAI model.
+
+[source,console]
+--------------------------------------------------
+PUT openai-embeddings
+{
+  "mappings": {
+    "properties": {
+      "content_embedding": { <1>
+        "type": "dense_vector", <2>
+        "dims": 1536 <3>
+      },
+      "content": { <4>
+        "type": "text" <5>
+      }
+    }
+  }
+}
+--------------------------------------------------
+<1> The name of the field to contain the generated tokens. It must be refrenced
+in the {infer} pipeline configuration in the next step.
+<2> The field to contain the tokens is a `dense_vector` field.
+<3> The output dimensions of the model. Find this value in the 
+https://platform.openai.com/docs/guides/embeddings/embedding-models[OpenAI documentation] 
+of the model you use.
+<4> The name of the field from which to create the sparse vector representation.
+In this example, the name of the field is `content`. It must be referenced in 
+the {infer} pipeline configuration in the next step.
+<5> The field type which is text in this example.
+
+
+[discrete]
+[[infer-openai-inference-ingest-pipeline]]
+==== Create an ingest pipeline with an inference processor
+
+Create an <<ingest,ingest pipeline>> with an
+<<inference-processor,{infer} processor>> to use the OpenAI model you referenced 
+in the {infer} task to infer against the data that is being ingested in the 
+pipeline.
+
+[source,console]
+--------------------------------------------------
+PUT _ingest/pipeline/openai_embeddings
+{
+  "processors": [
+    {
+      "inference": {
+        "model_id": "openai_embeddings", <1>
+        "input_output": { <2>
+          "input_field": "content",
+          "output_field": "content_embedding"
+        }
+      }
+    }
+  ]
+}
+--------------------------------------------------
+<1> The name of the inference task you created by using the 
+<<put-inference-api>>. 
+<2> Configuration object that defines the `input_field` for the {infer} process
+and the `output_field` that will contain the {infer} results.
+
+////
+[source,console]
+----
+DELETE _ingest/pipeline/openai_embeddings
+----
+// TEST[continued]
+////
+
+
+[discrete]
+[[infer-load-data]]
+==== Load data
+
+In this step, you load the data that you later use in the {infer} ingest
+pipeline to create embeddings from it.
+
+Use the `msmarco-passagetest2019-top1000` data set, which is a subset of the MS
+MARCO Passage Ranking data set. It consists of 200 queries, each accompanied by
+a list of relevant text passages. All unique passages, along with their IDs,
+have been extracted from that data set and compiled into a
+https://github.com/elastic/stack-docs/blob/main/docs/en/stack/ml/nlp/data/msmarco-passagetest2019-unique.tsv[tsv file].
+
+Download the file and upload it to your cluster using the
```

Comment:
I got a timeout uploading the file to the Data Visualizer. Let's create a smaller file for the purpose of this example. For readers following this example there's no need to process 60MB of data